util/unifyfs/src/unifyfs-rm.c

```diff
@@ -922,30 +1000,76 @@ int unifyfs_start_servers(unifyfs_resource_t* resource,
         fprintf(stderr, "failed to create server launch server process (%s)\n",
                         strerror(errno));
         return -errno;
-    } else if (pid == 0) {
+    }
+
+    args->arg_type = UNIFYFS_CONSTRUCT_ARGS_UNIFYFSD;
+    if (pid == 0) {
         if (args->script != NULL) {
-            return script_launch(resource, args);
+            unifyfsd_return_val = script_launch(resource, args);
         } else {
-            return resource_managers[resource->rm].launch(resource, args);
+            unifyfsd_return_val =
+	      resource_managers[resource->rm].launch(resource, args);
+        }
+    } else {
+        unifyfsd_return_val = wait_server_initialization(resource, args);
+        if (unifyfsd_return_val) {
+	    fprintf(stderr,
+		    "ERROR: failed to wait for server initialization: (%d s)\n",
+		    unifyfsd_return_val);
         }
     }
 
```

Comment:
We should be checking for a valid `unifyfsd_return_val` value here, and bailing if it is bad.  That will also allow you to get rid of the summation of return values at the end of the function:
```
return unifyfsd_return_val + transfer_return_val;
```